python/ModelTools.py

```diff
@@ -86,6 +86,9 @@ def __init__(self,datacard,options):
         ModelBuilderBase.__init__(self,options) 
         self.DC = datacard
         self.doModelBOnly = True
+        self.selfNormBins = []
+        self.extraNuisances = []
+        self.extraGlobalObservables = []
```

Comment:
shapeTools.py now creates additional bin-by-bin nuisance params and global observables, after ModelTools.py has already parsed the card are created the corresponding collections. These extra lists will be populated by shapeTools.py and then added to the pre-existing collections at the end.